parboiled-core/src/main/scala-3/org/parboiled2/support/OpTreeContext.scala

```diff
@@ -455,6 +455,34 @@ class OpTreeContext(parser: Expr[Parser])(using Quotes) {
       }
   }
 
+  private case class RunSubParser(fTree: Expr[_]) extends DefaultNonTerminalOpTree {
+    def ruleTraceNonTerminalKey = '{ RuleTrace.RunSubParser }
+    def renderInner(start: quoted.Expr[Int], wrapped: Boolean): Expr[Boolean] = {
+      def rewrite(arg: ValDef, tree: Term): Expr[Boolean] = {
+        tree match {
+          case Block(statements, res) => block(statements, rewrite(arg, res).asTerm).asExprOf[Boolean]
+          case Select(Apply(parserCons, List(consArg @ Ident(_)))), rule) if consArg.symbol == arg.symbol =>
```

Comment:
I added an extra check that the argument here is actually the one passed from the outside. Not so likely that it would be something else, but this way we can be sure...